File: catkin_ws/src/my_robot_tutorials/src/numberPubSub.cpp
#include <ros/ros.h>
#include <std_msgs/Int64.h>


std_msgs::Int64 value;

void cb_recieve_seed(const std_msgs::Int64 &seed)
{
    if(value.data==0)
    {
        ROS_INFO("Seed recieved: %i", (int)seed.data);
        value.data = seed.data;
    }
    else
    {
        value.data += 1;
        ROS_INFO("Incrementing counter %i", (int)value.data);
    }
}

void cb_recieve_count(const std_msgs::Int64 &count)
{
    ROS_INFO("Counter: %i", (int)count.data);
}

int main(int argc, char** argv)
{
    ros::init(argc, argv, "number_pubsub");
    ros::NodeHandle nh;
    ros::Subscriber sub_seed = nh.subscribe("/seed",
                                            1,
                                            cb_recieve_seed);

    ros::Subscriber sub_counter = nh.subscribe("/counter",
                                                1,
                                                cb_recieve_count);

    ros::Publisher pub = nh.advertise<std_msgs::Int64>("/counter", 1);

    ros::spin();

}

File: catkin_ws/src/my_robot_tutorials/scripts/numberPubSub.py
#! /usr/bin/env python

import rospy
from std_msgs.msg import Int64


value = 0


def cb_recieve_data(data:Int64):
    global value
    if not value:
        rospy.loginfo(f"Seed recieved {data.data}")
        value = data
    else:
        value.data += 1
        rospy.loginfo(f"Counter incrementing: {value.data}")
        

def cb_recieve_count(data:Int64):
    rospy.loginfo(f"Counter: {data.data}")

if __name__ == "__main__":
    rospy.init_node("number_pubsub")

    sub_seed = rospy.Subscriber("/seed", 
                           Int64, 
                           cb_recieve_data)
    
    sub_counter = rospy.Subscriber("/counter",
                                   Int64,
                                   cb_recieve_count)
    
    pub = rospy.Publisher("counter_count", Int64, queue_size=1)
    rate = rospy.Rate(1)

    while not rospy.is_shutdown():
        pub.publish(value)
        rate.sleep()
    rospy.loginfo("Node is exiting")